plugin/pkg/scheduler/api/types.go

```diff
@@ -128,24 +128,41 @@ type ExtenderConfig struct {
 	// HTTPTimeout specifies the timeout duration for a call to the extender. Filter timeout fails the scheduling of the pod. Prioritize
 	// timeout is ignored, k8s/other extenders priorities are used to select the node.
 	HTTPTimeout time.Duration
+	// NodeCacheCapable specifies that the extender is capable of caching node information,
+	// so the scheduler should only send minimal information about the eligible nodes
+	// assuming that the extender already cached full details of all nodes in the cluster
+	NodeCacheCapable bool
 }
 
 // ExtenderArgs represents the arguments needed by the extender to filter/prioritize
 // nodes for a pod.
 type ExtenderArgs struct {
 	// Pod being scheduled
 	Pod v1.Pod
-	// List of candidate nodes where the pod can be scheduled
-	Nodes v1.NodeList
+	// List of candidate nodes where the pod can be scheduled; to be populated
+	// only if extender is not capable of maintaining its own node cache
+	Nodes *v1.NodeList
```

Comment:
I think the backward compatibility concern might be that you changed the field from required to optional (Nodes to Nodes *) (At least I assume that's the implication of changing from Nodes to Nodes * -- I guess we're not using the JSON annotations here that say "omitempty" and such.)